### PR TITLE
build(datasets): Release `kedro-datasets` 1.7.0

### PR DESCRIPTION
**Pull request by ankatiyar**
## Description
Release `kedro-datasets` 1.7.0

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: kedro-datasets/RELEASE.md [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
+## Bug fixes and other changes
+## Community contributions
+
+# Release 1.7.0:
+## Major features and improvements
 * Added `polars.GenericDataSet`, a `GenericDataSet` backed by [polars](https://www.pola.rs/), a lightning fast dataframe package built entirely using Rust.
 
 ## Bug fixes and other changes
+* Fixed broken links in docstrings.
+* Revert pyspark support to 3.4+.
```
Comment (ankatiyar) [inline kedro-datasets/RELEASE.md:12]: Please suggest how to word this?

Comment (SajidAlamQB) [reply kedro-datasets/RELEASE.md:12]: Looks fine for the most part maybe:

```suggestion
* Reverted PySpark pin to <4.0.
```

**Comment by merelcht** [comment]
I'm glad to see the RTD build is catching the indentation error for the polars dataset, but kind of strange it didn't flag when that was merged? 🤔 @astrojuanlu

**Comment by ankatiyar** [comment]
> I'm glad to see the RTD build is catching the indentation error for the polars dataset, but kind of strange it didn't flag when that was merged? 🤔 @astrojuanlu

It wasn't added to the `.rst` file before.

**Comment by astrojuanlu** [comment]
> It wasn't added to the .rst file before.

Ugh, exactly, sorry about that. Should we try to make it automatic? Or at least add a checklist item in the pull request template.

**Comment by ankatiyar** [comment]
Finally fixed the docstring! Please review, @merelcht @astrojuanlu @SajidAlamQB!

**Review by SajidAlamQB — APPROVED** [review]

File: kedro-datasets/RELEASE.md [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
+## Major features and improvements
 * Added `polars.GenericDataSet`, a `GenericDataSet` backed by [polars](https://www.pola.rs/), a lightning fast dataframe package built entirely using Rust.
 
 ## Bug fixes and other changes
+* Fixed broken links in docstrings.
+* Reverted PySpark pin to <4.0.
+
 ## Community contributions
+Many thanks to the following Kedroids for contributing PRs to this release:
+* [Walber Moreira](https://github.com/wmoreiraa)
```
Comment (astrojuanlu) [inline kedro-datasets/RELEASE.md:16]: ❤️

**Review by astrojuanlu — COMMENTED** [review]
Thanks a lot for amending the Polars docstring 🙏🏽 may I ask to cherry-pick #324 for 1.7.0?

**Review by noklam — APPROVED** [review]

**Pull request merged**